### PR TITLE
Deprecate the `vscode` tasks and create the `devcontainer` ones

### PR DESCRIPTION
**Pull request by FlorentClarret**
### What does this PR do?

- Deprecate the `vscode` commands and replace them by the `devcontainer` ones
- Add new command to stop, start and restart the container.
- Add utility functions related to the devcontainer

### Motivation

- Devcontainers are not just for vscode. We can start them with the CLI or even other IDEs (I use Goland)
- We plan to use the devcontainers to run linters and unit tests on linux (for those on macOS)

### CODE REVIEW

File: tasks/vscode.py [hunk 1]
```diff
@@ -73,74 +46,15 @@ def setup_devcontainer(
...
-        print(f'{", ".join(build_tags[flavor].keys())} \n')
-        return
-
-    build_include = (
-        get_default_build_tags(build=target, arch=arch, flavor=flavor)
-        if build_include is None
-        else filter_incompatible_tags(build_include.split(","), arch=arch)
+    from tasks.devcontainer import setup
+
+    print('This command is deprecated, please use `devcontainer.setup` instead')
```
Comment (CelianR) [inline tasks/vscode.py:51]: Maybe we should output this to stderr + in red. We can add also `Running devcontainer.setup task` such that we know that it worked

File: tasks/devcontainer.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+
+@task
+def start(ctx, path="."):
+    """
+    Start the devcontainer
+    """
+    if not file().exists():
+        print("No devcontainer settings found.  Run `invoke devcontainer.setup` first.")
+        exit(1)
```
Comment (CelianR) [inline tasks/devcontainer.py:158]: Can you replace each `exit(1)` by this ?
```suggestion
        raise Exit(code=1)
```
And add `from invoke.exceptions import Exit`

File: tasks/devcontainer.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        json.dump(devcontainer, sf, indent=4, sort_keys=False, separators=(',', ': '))
+
+
+@task
+def start(ctx, path="."):
+    """
+    Start the devcontainer
+    """
+    if not file().exists():
+        print("No devcontainer settings found.  Run `invoke devcontainer.setup` first.")
```
Comment (CelianR) [inline tasks/devcontainer.py:157]: **🥜 nitpick:** ‏You can print to stderr + in red (same for the other `exit(1)` message)

**Review by CelianR — APPROVED** [review]
Some NITs, lgtm otherwise ! 👏

File: tasks/vscode.py (hunk 1 above)
Comment (FlorentClarret) [reply tasks/vscode.py]: Good idea, added!

File: tasks/devcontainer.py (hunk 3 above)
Comment (FlorentClarret) [reply tasks/devcontainer.py]: sure!

File: tasks/vscode.py [hunk 4]
```diff
@@ -73,80 +48,16 @@ def setup_devcontainer(
...
-        return
-
-    build_include = (
-        get_default_build_tags(build=target, arch=arch, flavor=flavor)
-        if build_include is None
-        else filter_incompatible_tags(build_include.split(","), arch=arch)
+    from tasks.devcontainer import setup
+
+    print(color_message('This command is deprecated, please use `devcontainer.setup` instead', "orange"))
+    print("Running `devcontainer.set_buildtags`...")
```
Comment (FlorentClarret) [inline tasks/vscode.py:54]:
```suggestion
    print("Running `devcontainer.setup`...")
```

File: tasks/devcontainer.py [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+def set_buildtags(
+    _,
+    target="agent",
+    build_include=None,
+    build_exclude=None,
+    flavor=AgentFlavor.base.name,
+    arch='x64',
+):
+    """
+    Modifies devcontainer settings file for this project to include correct build tags
```
Comment (KevinFairise2) [inline tasks/devcontainer.py:35]: I think it only modifies `.vscode/settings.json` which can be unrelated to devcontainer

File: tasks/install_tasks.py [hunk 6]
```diff
@@ -85,3 +85,11 @@ def install_shellcheck(ctx, version="0.8.0", destination="/usr/local/bin"):
...
     ctx.run(f"cp \"/tmp/shellcheck-v{version}/shellcheck\" {destination}")
     ctx.run(f"rm -rf \"/tmp/shellcheck-v{version}\"")
+
+
+@task
+def install_devcontainer_cli(ctx):
+    """
+    Install the devcontainer CLI
+    """
+    ctx.run("npm install -g @devcontainers/cli")
```
Comment (KevinFairise2) [inline tasks/install_tasks.py:95]: It assumes `npm` is installed do you know if it is installed by default with the Datadog laptop setup or should we maybe give info on how to install it if needed?

**Review by KevinFairise2 — COMMENTED** [review]
Some small comments. Maybe we could keep the `vscode.set_buildtags` in `vscode.py. Because if I understand correctly it can be used without devcontainer

File: tasks/install_tasks.py (hunk 6 above)
Comment (FlorentClarret) [reply tasks/install_tasks.py:95]: Yep that's installed by default on our laptops

File: tasks/devcontainer.py (hunk 5 above)
Comment (FlorentClarret) [reply tasks/devcontainer.py]: you're right, it has nothing to do with containers 🤦

File: tasks/vscode.py [hunk 7]
```diff
@@ -30,34 +21,18 @@ def set_buildtags(
...
-        print(f'{", ".join(build_tags[flavor].keys())} \n')
-        return
-
-    build_include = (
-        get_default_build_tags(build=target, arch=arch, flavor=flavor)
-        if build_include is None
-        else filter_incompatible_tags(build_include.split(","), arch=arch)
+    from tasks.devcontainer import set_buildtags
+
+    print(color_message('This command is deprecated, please use `devcontainer.set_buildtags` instead', "orange"))
```
[the remark this answers is not in the document]
Comment (FlorentClarret) [reply tasks/vscode.py]: addressed! thanks

**Review by KevinFairise2 — APPROVED** [review]

**Pull request merged**